.github/workflows/wheels-dependencies.sh

```diff
@@ -173,12 +168,7 @@ function build {
     fi
     build_new_zlib
 
-    ORIGINAL_LDFLAGS=$LDFLAGS
-    if [[ -n "$IS_MACOS" ]] && [[ "$CIBW_ARCHS" == "arm64" ]]; then
-        LDFLAGS="${LDFLAGS} -ld64"
-    fi
```

Comment:
I'm a little wary of removing this. It was a workaround for fdintino/pillow-avif-plugin#59. I wrote this comment elsewhere summarizing my findings:

https://github.com/image-rs/image/issues/2108#issuecomment-2289412699

>> [...] I encountered an illegal instruction issue with libavif that I ultimately diagnosed as a bug in Apple's new(ish) static linker. Stepping through the executable in lldb you could see 4 bytes inserted into the middle of a subroutine that were invalid assembly (so, not even a valid but unsupported ARM intrinsic). If those 4 bytes were removed, the subroutine ran correctly. I noticed that the issue went away if I compiled with `-O2`; my best guess is that the bug occurs in functions with both inline assembly and C code that clang optimizes with ARM intrinsics (I imagine these are necessary but not sufficient conditions, or else people would be encountering many more corrupt executables). I was able to reliably avoid the issue if I compiled with the old ld64 linker, by adding `-ld64` to `LDFLAGS`.

The test `test_aom_optimizations` served as a regression test of sorts, and it's no longer failing. So perhaps the linker flag has since been fixed and we no longer need this flag.